### DISCUSSION TITLE
picoev: simpler way to get header values

### DISCUSSION THREAD

**Discussion by duselguy** [vlang/v#11862] [Ideas]
**V version:** 
V 0.2.2 2c4a59f

**OS:** 
20.04.1-Ubuntu

**What did you do?**
Try to find a simple way to get the header value by the header name in request 

**What did you expect to see?**
Code snippet for this.

**Comment by duselguy** [comment]
I couldn't find a better one than:
`for i in 0 .. req.num_headers {
      unsafe {
          x := req.headers[i].name.vstring_with_len(req.headers[i].name_len)
          y := req.headers[i].value.vstring_with_len(req.headers[i].value_len)
          println(x + ' is ' + y)
       }
 }
`

**Comment by Terisback** [comment]
I have ideas for that

**Comment by dumblob** [comment]
@Terisback there were some ideas about [lazy parsing](https://github.com/vlang/v/issues/8986#issuecomment-787063058 ) but so far nobody designed any API around that (which is a shame as it provably offers a significant performance benefit in practice).

**Comment by duselguy** [comment]
@dumblob
I hope greedy parsing does not create (performance) problems for modern single page applications.

**Comment by dumblob** [comment]
@duselguy I bet it does (it's not just about reading and copying the data to structures, but it's also about malloc(), free() versus stack etc.).

**Comment by dumblob** [comment]
Some real-life inspiration how to do "lazy/streaming parsing" at scale: https://github.com/cloudflare/lol-html (from Cloudflare).
